subprojects/build-cache-http/src/main/java/org/gradle/caching/http/internal/HttpBuildCacheService.java

```diff
@@ -195,4 +193,15 @@ private static URI safeUri(URI uri) {
             throw UncheckedException.throwAsUncheckedException(e);
         }
     }
+
```

Comment:
I think it is worth adding documentation here:
```suggestion

    /**
     * Add a trailing slash to the given URI's path if necessary.
     *
     * @param uri the original URI
     * @return a URI guaranteed to have a trailing slash in the path
     */
```